src/inmanta/data/dataview.py

```diff
@@ -486,68 +508,70 @@ def get_extra_url_parameters(self) -> dict[str, str]:
-                cte AS (
-                    /* Initial row for recursion: select relevant version for first resource */
-                    ( {subquery_latest_version_for_single_resource(higher_than=None)} )
-                    UNION ALL
-                    SELECT next_r.*
-                    FROM cte curr_r
-                    CROSS JOIN LATERAL (
-                        /* Recurse: select relevant version for next resource (one higher in the sort order than current) */
-                        {subquery_latest_version_for_single_resource(higher_than="curr_r.resource_id")}
-                    ) next_r
+                ), versioned_resources AS (
+                    SELECT
+                        rps.*,
+                        CASE
+                            -- TODO: first branch may be redundant
+                            WHEN EXISTS (
+                                SELECT 1
+                                FROM resource AS r
+                                WHERE r.environment = rps.environment AND r.resource_id = rps.resource_id AND r.model = (
+                                    SELECT version FROM latest_version
+                                )
+                            ) THEN (SELECT version FROM latest_version)
+                            ELSE (
+                                SELECT MAX(r.model)
+                                FROM resource AS r
+                                JOIN configurationmodel AS m
+                                    ON r.environment = m.environment AND r.model = m.version AND m.released = TRUE
+                                WHERE r.environment = rps.environment AND r.resource_id = rps.resource_id
+                                GROUP BY r.resource_id
+                            )
+                        END AS version
+                    FROM resource_persistent_state AS rps
+                    WHERE rps.environment = $1
+                ), result AS (
+                    SELECT
+                        r.resource_id,
+                        r.attributes,
+                        r.resource_type,
+                        r.agent,
+                        r.resource_id_value,
+                        r.model,
+                        r.environment,
+                        (
+                            CASE
+                                WHEN r.model < (SELECT version FROM latest_version)
+                                    THEN 'orphaned'
+                                WHEN (r.status = 'deploying')
+                                    THEN r.status::text
+                                ELSE
+                                    rps.last_non_deploying_status::text
+                            END
+                        ) as status
+                    FROM versioned_resources AS rps
+                    -- TODO: is this accurate? Greatly improves COUNT(*) performance, may slightly degrade sort performance
+                    -- TODO: LEFT iff including orphans
```

Comment:
These TODO's can go now. And perhaps we should add a comment
```suggestion
-- LEFT join for trivial `COUNT(*)`. Not applicable when filtering orphans because left table contains orphans.
```